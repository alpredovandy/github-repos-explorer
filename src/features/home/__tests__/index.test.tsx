// import { render } from '@testing-library/react';
// import { describe, expect, test, vi } from 'vitest';

// import { MOCK_GIFS } from '@/domains/__mocks__/Gif.ts';
// import { CatalogProps } from '@/features/home/sections/types';

// import Home from '..';

// const mockCatalog = vi.fn();
// vi.mock('@/features/home/sections/Catalog', () => ({
//     default: (props: CatalogProps) => {
//         mockCatalog(props);
//         return <div data-testid="home-catalog" {...props} />;
//     },
// }));

// vi.mock('@tanstack/react-query', () => ({
//     useQuery: () => ({
//         data: MOCK_GIFS,
//         isPending: false,
//     }),
// }));

// describe('@/modules: Home', () => {
//     test('Snapshot', () => {
//         const { container } = render(<Home />);

//         expect(container).toMatchSnapshot();
//     });
// });
