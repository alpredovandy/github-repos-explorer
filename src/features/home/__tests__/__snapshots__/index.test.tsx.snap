// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@/modules: Home > Snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-5 lg:w-2/3"
  >
    <div
      class="flex justify-between items-center"
    >
      <span
        class="text-md font-bold text-teal-400"
      >
        Powered by Giphy.com
      </span>
      <span
        class="italic text-amber-400"
      >
        @alpredovandy
      </span>
    </div>
    <label
      class="input input-bordered flex items-center"
    >
      <input
        class="grow"
        placeholder="Search..."
        type="text"
        value=""
      />
      <svg
        height="24px"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m19.6 21l-6.3-6.3q-.75.6-1.725.95T9.5 16q-2.725 0-4.612-1.888T3 9.5t1.888-4.612T9.5 3t4.613 1.888T16 9.5q0 1.1-.35 2.075T14.7 13.3l6.3 6.3zM9.5 14q1.875 0 3.188-1.312T14 9.5t-1.312-3.187T9.5 5T6.313 6.313T5 9.5t1.313 3.188T9.5 14"
          fill="currentColor"
        />
      </svg>
    </label>
    <div
      class="tabs tabs-boxed"
      role="tablist"
    >
      <a
        class="tab tab-active"
        role="tab"
      >
        Popular
      </a>
      <a
        class="tab "
        role="tab"
      >
        Trending
      </a>
    </div>
    <div
      data-testid="home-catalog"
      gifs="[object Object]"
    />
  </div>
</div>
`;
